### DISCUSSION TITLE
Can you explain why my code dosnt work please ?!

### DISCUSSION THREAD

**Discussion by WeissLuxford** [codewars/codewars.com#2750] [Q&A] [answered]
Pls, help me? whats wrong with my code, i have no idea understand why it doesn't work it passes (50% of test) and on other tests catches this error Value is not what was expected I dont, know what should i change
```
function combine(a = {}, b = {}, c = {}) {
  const arrObj = [a, b, c]
  const newObj = {}
  arrObj.forEach((value, index, arr) => {
    for (const [keys, values] of Object.entries(value)) {
      if( !newObj.hasOwnProperty(keys)){
        newObj[keys] = values
      } else {
        newObj[keys] = newObj[keys] + values
      }
    }
  })
  console.log(arrObj)
  console.log(newObj)
  return newObj
}
```

in this kata [Combine Objects](https://www.codewars.com/kata/combine-objects)

**Comment by flaviorcb** [comment, marked as answer]
The problem is : the input can have more than 3 objects.
"Your task is to write a function that takes two or more objects"

**Comment by Daringu** [comment]
the function has to be able to accept more than 2 two arguments(2,3,4,5,6,7...).How can you do it?Use "rest" in your arguments to collect them into one array and then you will be able to iterate in the array.

**Comment by Daringu** [comment]
also remember about reduce method.It's very useful method

**Reply by WeissLuxford** [reply]
Thanks, for your advice, ill remember, But I still need to fully understand how ```reduce``` works.

**Comment by Oahaijil** [comment]
The problem is : the input can have more than 3 objects.
"Your task is to write a function that takes two or more objects"

**Comment by hobovsky** [comment]
Githib discussions are meant for discussing matters related to design and features of Codewars platform, and not for help with individual kata. If you need a help with a kata, you can ask your questions in `#help-solve` channel of [Codewars Discord](https://discord.gg/mSwJWRvkHA).
